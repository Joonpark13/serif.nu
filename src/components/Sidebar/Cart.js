import React from 'react';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';

export const styles = {
  cartHeading: {
    padding: '16px',
  },
};

function Cart({ classes, sections }) {
  const label = sections.length === 1 ? 'Class' : 'Classes';
  return (
    <Typography variant="h5" className={classes.cartHeading}>
      {sections.length}
      {' '}
      {label}
    </Typography>
  );
}

Cart.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  sections: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export { Cart as UnstyledCart };
export default withStyles(styles)(Cart);
