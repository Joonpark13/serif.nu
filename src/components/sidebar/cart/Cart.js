import React from 'react';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/styles';
import PropTypes from 'prop-types';
import CartSection from './CartSection';

export const styles = {
  cartHeading: {
    padding: '16px',
  },
};

function Cart({ classes, sections }) {
  const uniqueSections = [];
  const usedIDs = [];

  sections.forEach(
    (section) => {
      if (!usedIDs.includes(section.id)) {
        usedIDs.push(section.id);
        uniqueSections.push(section);
      }
    },
  );

  const label = uniqueSections.length === 1 ? 'Class' : 'Classes';

  return (
    <React.Fragment>
      <Typography variant="h5" className={classes.cartHeading}>
        {uniqueSections.length}
        {' '}
        {label}
      </Typography>
      {uniqueSections.map(section => <CartSection key={section.id} section={section} />)}
    </React.Fragment>
  );
}

Cart.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  sections: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export { Cart as UnstyledCart };
export default withStyles(styles)(Cart);
