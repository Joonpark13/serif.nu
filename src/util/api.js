/* eslint-disable import/prefer-default-export */
export function searchURL(query) {
  return `${process.env.API_URL}/search?term_id=4720&search_query=${query}`;
}

export function sectionsURL(termId, schoolAbbv, subjectAbbv, courseAbbv) {
  return `${process.env.API_URL}/sections?term_id=${termId}&school_abbv=${schoolAbbv}&subject_abbv=${subjectAbbv}&course_abbv=${courseAbbv}`;
}
