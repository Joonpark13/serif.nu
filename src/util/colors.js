export const northwesternPurple = '#4E2A84';
export const northwesternBrightOrange = '#FFC520';
export const northwesternRichBlack10 = '#D8D6D6';
