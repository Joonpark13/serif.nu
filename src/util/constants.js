/* eslint-disable import/prefer-default-export */
export const CURRENT_TERM = '4730';
