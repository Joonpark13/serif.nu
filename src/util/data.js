import termsData from 'data/terms.json';

/* eslint-disable import/prefer-default-export */
/* istanbul ignore next */
export const currentTerm = termsData.reduce((highestTerm, term) => {
  if (term.id > highestTerm.id) {
    return term;
  }
  return highestTerm;
});
/* eslint-enable import/prefer-default-export */
