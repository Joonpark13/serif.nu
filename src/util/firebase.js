import firebase from 'firebase/app';
import 'firebase/firestore';
import 'firebase/auth';

firebase.initializeApp({
  apiKey: 'AIzaSyDluAG6bnqleqJ3yjPvGdlb3RyZP3U5JIQ',
  authDomain: 'sans-serif-northwestern.firebaseapp.com',
  databaseURL: 'https://sans-serif-northwestern.firebaseio.com',
  projectId: 'sans-serif-northwestern',
  storageBucket: 'sans-serif-northwestern.appspot.com',
  messagingSenderId: '1078100635889',
});

export const db = firebase.firestore();
export const auth = firebase.auth();
