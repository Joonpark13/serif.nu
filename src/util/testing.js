/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';
import * as useSelector from './use-selector';

jest.mock('./use-selector');

// Helper function that returns a function which creates enzyme wrappers with default props
export function wrapperCreator(Component, defaultProps = {}) {
  let classes;
  return props => shallow(<Component classes={classes} {...defaultProps} {...props} />);
}

/* istanbul ignore next */
export function mockUseSelector(...args) {
  useSelector.default.mockReset();
  args.forEach(arg => useSelector.default.mockReturnValueOnce(arg));

  return useSelector.default;
}

/* istanbul ignore next */
export function mockUseDispatch() {
  const dispatchSpy = jest.fn();
  reactRedux.useDispatch.mockReturnValue(dispatchSpy);
  return dispatchSpy;
}
