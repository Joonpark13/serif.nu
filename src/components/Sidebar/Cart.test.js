import React from 'react';
import { shallow } from 'enzyme';
import { mockStyles } from 'util/testing';
import { UnstyledCart, styles } from './Cart';

describe('Cart', () => {
  it('renders correctly', () => {
    const classes = mockStyles(styles);
    const wrapper = shallow(<UnstyledCart sections={[]} classes={classes} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly with one section', () => {
    const classes = mockStyles(styles);
    const wrapper = shallow(<UnstyledCart sections={[{}]} classes={classes} />);

    expect(wrapper).toMatchSnapshot();
  });
});
