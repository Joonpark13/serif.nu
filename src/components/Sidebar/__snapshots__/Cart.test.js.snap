// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cart
    classes={
      Object {
        "cartHeading": "",
      }
    }
    sections={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        0,
        " ",
        "Classes",
      ],
      "className": "",
      "variant": "h5",
    },
    "ref": null,
    "rendered": Array [
      0,
      " ",
      "Classes",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          0,
          " ",
          "Classes",
        ],
        "className": "",
        "variant": "h5",
      },
      "ref": null,
      "rendered": Array [
        0,
        " ",
        "Classes",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Cart renders correctly with one section 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cart
    classes={
      Object {
        "cartHeading": "",
      }
    }
    sections={
      Array [
        Object {},
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        1,
        " ",
        "Class",
      ],
      "className": "",
      "variant": "h5",
    },
    "ref": null,
    "rendered": Array [
      1,
      " ",
      "Class",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          1,
          " ",
          "Class",
        ],
        "className": "",
        "variant": "h5",
      },
      "ref": null,
      "rendered": Array [
        1,
        " ",
        "Class",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
